code/core/src/preview-api/modules/preview-web/render/StoryRender.ts

```diff
@@ -196,22 +203,28 @@ export class StoryRender<TRenderer extends Renderer> implements Render<TRenderer
         step: (label, play) => runStep(label, play, context),
         context: null!,
         canvas: {} as Canvas,
-        mount: null!,
         renderToCanvas: async () => {
+          const teardown = await this.renderToScreen(renderContext, canvasElement);
+          this.teardownRender = teardown || (() => {});
+          mounted = true;
+        },
+        // The story provides (set in a renderer) a mount function that is a higher order function
+        // (context) => (...args) => Canvas
+        //
+        // Before assigning it to the context, we resolve the context dependency,
+        // so that a user can just call it as await mount(...args) in their play function.
+        mount: async (...args) => {
+          let mountReturn: Awaited<ReturnType<StoryContext['mount']>> = null!;
           await this.runPhase(abortSignal, 'rendering', async () => {
-            const teardown = await this.renderToScreen(renderContext, canvasElement);
-            this.teardownRender = teardown || (() => {});
-            mounted = true;
+            mountReturn = await story.mount(context)(...args);
           });
-
-          if (isMountDestructured) {
-            // put the phase back to playing if mount is used inside a play function
-            await this.runPhase(abortSignal, 'playing', async () => {});
-          }
+          // start playing phase if mount is used inside a play function
+          if (isMountDestructured) await this.runPhase(abortSignal, 'playing');
+          return mountReturn;
         },
       };
```

Comment:
Before we did the phase change in renderToCanvas.

But it actually needs to be done here. As `renderToCanvas` "fixture" might not be used inside of the `mount` "fixture". In theory, the `mount` fixture can do everything itself, although that is not how it is currently implemented in our own renderers.